rg_shipit/classes/ShipitIntegrationCore.php

```diff
@@ -0,0 +1,159 @@
+
+    public function __construct($email, $token, $version) {
+      $this->base = 'http://api.shipit.cl/v';
+      $this->email = $email;
+      $this->token = $token;
+      $this->headers = array(
+        'Content-Type' => 'application/json',
+        'X-Shipit-Email' => $email,
+        'X-Shipit-Access-Token' => $token,
+        'Accept' => 'application/jsonapplication/vnd.shipit.v' . $version
+      );
+    }
+
+    function packages($package = array()) {
+      $client = new ShipitHttpClient($this->base.'/packages', $this->headers);
+      $response = $client->post($package);
+      if ($response->getStatusCode() != 200) {
+        ShipitTools::log('PrestaShop ('._PS_VERSION_.'), package response: '.print_r($response,true));
+      } else {
+        $package = json_decode($response['body']);
+      }
+      return $package;
+    }
+
+    function massivePackages($packages = array()) {
+      $client = new ShipitHttpClient($this->base.'/packages/mass_create', $this->headers);
+      $response = $client->post($packages);
+      if ($response->getStatusCode() != 200) {
+        ShipitTools::log('PrestaShop ('._PS_VERSION_.'), massive package response: '.print_r($response,true));
+      } else {
+        $package = json_decode($response['body']);
+      }
+      return $package;
+    }
+
+    function orders($order = array()) {
+      $client = new ShipitHttpClient($this->base.'/orders', $this->headers);
+      $response = $client->post($order);
+      if ($response->getStatusCode() != 200) {
+        ShipitTools::log('PrestaShop ('._PS_VERSION_.'), orders response: '.print_r($response,true));
+      } else {
+        $order = json_decode($response['body']);
+      }
+      return $order;
+    }
+
+    function shipments($shipment) {
+      $client = new ShipitHttpClient($this->base.'/shipments', $this->headers);
+      $response = $client->post(['shipment' => $shipment]);
+      if ($response->getStatusCode() != 200) {
+        ShipitTools::log('PrestaShop ('._PS_VERSION_.'), shipment response: '.print_r($response,true));
+      } else {
+        $shipment = json_decode($response->getBody());
+      }
+      return $shipment->id;
+    }
+
+    function administrative() {
+      $client = new ShipitHttpClient($this->base.'/setup/administrative', $this->headers);
+      $response = $client->get();
+      if ($response->getStatusCode() != 200) {
+        ShipitTools::log('PrestaShop ('._PS_VERSION_.'), administrative response: '.print_r($response,true));
+      } else {
+        $company = json_decode($response['body']);
+      }
+      return $company;
+    }
+
+    function communes() {
+      $client = new ShipitHttpClient($this->base.'/communes', $this->headers);
+      $response = $client->get();
+      if ($response->getStatusCode() != 200) {
+        ShipitTools::log('PrestaShop ('._PS_VERSION_.'), communes response: '.print_r($response,true));
+      } else {
+        $communes = json_decode($response->getBody());
```

Comment:
why are you using `getBody()` and in other cases `['body']` (asking cz I'm not familiarized with this)... what I'm seeing here is the probability of reuse a lot of code and get more elasticity with the api calls.

maybe generate an abstract method that receives something like: `($method, $endpoint, $params, $entity)` you will be able to just call one method and that gives you less issues nodes at api call moment.

Reply:
it should be with the getBody method, the other services I haven't used, that's why they are taking from an array, but I'll update